scripts/release.mjs

```diff
@@ -4,20 +4,32 @@
 import { cwd, env } from 'process';
 import { createRequire } from 'module';
 
-import { $, argv, question } from 'zx';
+import { $, question } from 'zx';
 
 const moduleMetaUrl = import.meta.url;
 const require = createRequire(moduleMetaUrl);
 let { NPM_TOKEN } = env;
-let { DIST_TAG, BUMP } = argv;
+let BUMP = 'patch';
+let DIST_TAG = 'latest';
 let twoFactorCode;
 
-DIST_TAG = DIST_TAG || 'latest';
-BUMP = BUMP || 'patch';
 NPM_TOKEN = NPM_TOKEN || '';
 
+// This will determine the type of release based on the git branch. When the default branch is used, it will be a `patch` that's published to npm under the `latest` dist-tag. Any other branch will be a `prelease` that's published to npm under the `alpha` dist-tag.
+
+let { stdout: CURRENT_BRANCH } = await $`git rev-parse --abbrev-ref HEAD`;
+CURRENT_BRANCH = CURRENT_BRANCH.trim();
+let { stdout: DEFAULT_BRANCH } = await $`git remote show origin | sed -n '/HEAD branch/s/.*: //p'`;
+DEFAULT_BRANCH = DEFAULT_BRANCH.trim();
+
 await $`grabthar-validate-git`;
 await $`grabthar-validate-npm`;
+
+if (CURRENT_BRANCH !== DEFAULT_BRANCH) {
```

Comment:
Should we only rely on branch name to determine if its a candidate for alpha or latest tag 🤔 ? I am concerned about someone using default branch on their fork while working.

Reply:
@ravishekhar, our overarching goal is to no longer have to manage npm package permissions for individual contributors, and, instead, use CI to publish. That being said, when using GitHub Actions, it isn't possible to run workflows on forked repos:

![Screen Shot 2022-03-18 at 9 10 37 AM](https://user-images.githubusercontent.com/20399044/159048389-e4160cbf-c6a3-4a6a-88be-5d4882ac093e.png)

Once we are done making changes to our release workflows, we'll likely remove individual contributors from npm packages. This will prevent anyone without permissions to publish locally from the default branch of a forked repo (cc: @gregjopa).

Reply:
@ravishekhar, I just pushed a commit that uses the GitHub Repos API to check if a repo is forked, and if it is, it throws an error. Please take a look and let me know what you think.